netutils/usrsock_rpmsg/usrsock_rpmsg_client.c

```diff
@@ -155,6 +161,41 @@ static int usrsock_rpmsg_ept_cb(struct rpmsg_endpoint *ept, void *data,
   return ret;
 }
 
+#ifdef CONFIG_NETDEV_WIRELESS_IOCTL
+
+static void usersock_coherent_cache(FAR void *buf,
+                                    enum usrsock_cache_action_e action)
+{
+  FAR struct usrsock_request_ioctl_s *req = buf;
+  FAR struct iwreq *wlreq;
+
+  if (req->head.reqid == USRSOCK_REQUEST_IOCTL)
+    {
+      if (WL_IS80211POINTERCMD(req->cmd))
+        {
+          wlreq = (FAR struct iwreq *)(req + 1);
+          if (action == USRSOCK_COHERENT_BEFORE)
+            {
+              metal_cache_flush(wlreq->u.data.pointer,
```

Comment:
Is `metal_` some arch specific API. Maybe we can use `up_invalidate_dcache`/`up_flush_dcache` instead?

Reply:
usrsock rpmag is built on the top of openamp/libmetal, it's reasonable to call meta_ API instead of NuttX specific API to improve the portability.